src/core.rs

```diff
@@ -307,11 +318,9 @@ impl Fragment for Word<'_> {
         self.width as f64
     }
 
-    // We assume the whitespace consist of ' ' only. This allows us to
-    // compute the display width in constant time.
```

Comment:
This might be important... Have you tried

```sh
$ cd  benchmarks
$ cargo criterion fill_optimal_fit_unicode
```

and checked the graph in `target/criterion/reports/String%20lengths/fill_optimal_fit_unicode/index.html`?

It should look something like this so that we know that performance scales linearly:

![image](https://user-images.githubusercontent.com/89623/200168950-dd25759a-39a6-4a78-b6c7-17992f1ebd26.png)

Reply:
I think the change here might be just fine: most `Word` values will have a single space as their whitespace, so the performance will still be linear (at least with the simple benchmark I've been using).

Also, I think I'll finally rename the benchmark to something else than `String lengths` :smile:

Reply:
These are the results as of a901fce7b0165ba2186aad3e5fdca02b96976949:

![chart](https://user-images.githubusercontent.com/36740602/200991180-35550fd8-819c-451c-a8c5-c1d74ad83ffa.png)

So it looks like things are ok. My chart looks a little different than yours though, but I think it's mainly due to my default font being different.

Reply:
Cool, that looks all good to me.